airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -1257,7 +1265,11 @@ def generate_job(self):
         :return: Dict representing Dataproc job
         """
         self.create_job_template()
+        if self.job_template is None:
+            raise AirflowException('Job should be set here')
```

Comment:
What users should do seeing this error? Is it even possible to reach this part?

Reply:
No. But I have no idea how to make it in the way that MyPy does not complain. It's just badly structured I think - it should not store the job as self  but create_job_template should rather return the templte which is "not None". But I think this might be a separate refactor when the time comes.

Reply:
Actually - maybe I CAN fix it now.

Reply:
Yeah. It's much better now actually, without breaking backwards compatibility!